core/arch/interp.c

```diff
@@ -4196,6 +4223,12 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
         }
 #endif
     }
+    else if (dcontext->forged_exception_addr == bb->start_pc) {
+        /* Might have been cleared by client_process_bb. */
```

Comment:
Ok, I think it is best not to mess up the app.

But is it not possible to let the client add different exits to this basic block, but to keep the default fall through to the special exit where we forge the single step exception ?